system_setup/client/client.py

```diff
@@ -38,16 +39,10 @@ class SystemSetupClient(SubiquityClient):
         ]
 
     def __init__(self, opts):
-        # TODO WSL:
-        # 1. remove reconfigure flag
-        # 2. decide on which UI to show up based on existing user UID >=1000
-        #    (or default user set in wsl.conf?)
-        # 3. provide an API for this for the flutter UI to know about it
-        # 4. Add Configuration Base page before Advanced
-        # 5. Add language page
-        # self.variant = "wsl_configuration"
-        if opts.reconfigure:
+        if is_reconfigure(opts.dry_run):
```

Comment:
We shouldn’t detect reconfiguration mode in the client. The model is basically:

1. Let the server detects based on UID rules if we are in setup or reconfiguration mode.
2. Get it from the server API in the client in which mode we are
3. A little bit silly with the current implementation, but needed… Post the mode back to the server for it to load the expected server models…

Even if this seems counter-intuitive, because of the 3. this allows:
- the server to do the initial detection, as this is where the actual code is running (client could be remote)
- let the client override it if needed (for the tests for instance)

Reply:
I am not sure how should I do this since we should then call an async function in __init__, which of course crashes. What is the proper way I should implement this?

Reply:
This is a good question as this is triggered by the fact of having static class member with the base inherited/+meta class making magic on controllers, replacing the objects. I think of multiple hackish solutions, but maybe @mwhudson or @dbungert has a more upstream-compatible idea?

Reply:
So I guess there are two broad approaches here.

1. Have all controllers loaded in the client always but have the ones that are inappropriate for the current mode skip themselves (the generic tui client code has support for this, just raise subiquitycore.tuicontroller.Skip from make_ui)
2. Change the set of controllers to be loaded after seeing what the server says

The first is probably easier, tbh, although the second might be aesthetically more pleasing. Nothing in the installer tui client touches the controllers instance until the first status API call has returned so while the ControllerSet class wasn't written with this in mind it looks like assigning `controller_names` between the instance being constructed and calling `load_all` will work fine (another approach would be to construct the ControllerSet later, probably).

Reply:
WRT the POSTing the variant back... surely something in the system_state server startup code can call set_source_variant on the model during startup?

Reply:
Updated, and tested locally. It should now work properly.